src/components/AudioTrack/layouts/Full.vue

```diff
@@ -52,9 +54,30 @@ export default {
       }
       return '--:--'
     }
-
+    /**
+     * Creates a list of { extension_name, download_url } objects
+     * for DownloadButton
+     * @param {AudioDetail} audio
+     */
+    const getFormats = (audio) => {
+      // TODO: finalize the property names in
+      //  formats objects and alt_files objects
+      let formats = [
+        { extension_name: audio.filetype, download_url: audio.url },
+      ]
```

Comment:
If the `audio` object already uses `filetype` and `url` for the properties I think we should just update the download button component to accept those properties, so we can just pass a list of audio objects instead.

The only concern is whether `filetype` is a truly human readable filetype (like it needs to be `MP3 96kbs` instead of `mp31` for example)

Reply:
Yeah we might want to keep an enum of human readable file name mappings here, or more likely push this down to the catalog level and prettify the `format` before saving the rows.

Reply:
>  I separated the format name from the URL because I assumed the true format name couldn't be derived from some download URLs (this is an example). We should have an MP3 96kbs and an MP3 V0 for jamendo for each of their available MP3 download formats.

There is no _standard_ way of deriving the audio format from the URL: for example, Jamendo has format information after the last `/`, and Wikimedia often has it as an extension on the download URL. This is why we want to add `filetype` as a column to the Catalog database.